hibernate-core/src/main/java/org/hibernate/hql/internal/ast/tree/BinaryLogicOperatorNode.java

```diff
@@ -206,6 +219,10 @@ protected void translate(
 		}
 	}
 
+	private boolean isInsideSetClause() {
+		return getWalker().getCurrentClauseType() == HqlSqlTokenTypes.SET;
+	}
+
```

Comment:
feel free to Ignore my comment for I am not sure what I am saying, :). Usually `INSERT` and `SET` follow the same or similar syntax. Do your code changes fix the insert scenario, btw? If not, should we include `INSERT` above?

Reply:
HQL INSERT does not follow the same syntax used for the SET https://docs.jboss.org/hibernate/orm/5.4/userguide/html_single/Hibernate_User_Guide.html#_hql_syntax_for_insert.

Reply:
I meant `insert into ... values (...)`